### PR TITLE
chia/cmds: separate model from view in chia farm summary

### PR DESCRIPTION
**Pull request by 7AC**
This is a first step towards making the data accessible via an API.
I'm already using a rougher version of this that is parsing the `chia farm summary`
output in a web server and rendering the data as JSON for Grafana.

Somewhat related to https://github.com/Chia-Network/chia-blockchain-gui/issues/266 as well.

### CODE REVIEW

File: chia/cmds/farm_funcs.py [hunk 1]
```diff
@@ -181,69 +181,101 @@ async def challenges(farmer_rpc_port: int, limit: int) -> None:
...
+        self.user_transaction_fees = None
+        self.block_rewards = None
+        self.last_height_farmed = None
+        self.total_plot_size = None
+        self.plot_count = None
+        self.estimated_network_space = None
+        self.expected_time_to_win = None
+
+    def __str__(self):
+        def format(name: str, value: str) -> str:
```
Comment (Nikolaj-K) [inline chia/cmds/farm_funcs.py:197]: I note that it doesn't seem like it's expected that the inputs are actually all string.

File: chia/cmds/farm_funcs.py [hunk 2]
```diff
@@ -181,69 +181,101 @@ async def challenges(farmer_rpc_port: int, limit: int) -> None:
...
+        output += "Total size of plots: "
+        if self.total_plot_size is None:
+            output += "Unknown"
+        else:
+            plots_space_human_readable = self.total_plot_size / 1024 ** 3
+            unit = "GiB"
+            if plots_space_human_readable >= 1024 ** 2:
+                plots_space_human_readable = plots_space_human_readable / (1024 ** 2)
+                unit = "PiB"
+            elif plots_space_human_readable >= 1024:
```
Comment (Nikolaj-K) [inline chia/cmds/farm_funcs.py:222]: You could also get rid of the elif in favour of a second if and thus drop the two **2's

**Comment by Nikolaj-K** [comment]
@7AC Hello I hope this comes back in some form? :)

**Comment by 7AC** [comment]
> @7AC Hello I hope this info maybe comes back in some form? :)

Yes definitely, thank you for the review! I also have subsequent commits for the http server and a Prometheus exporter, I just need to polish everything (and address your comments).

**Pull request closed without merging**